store/microBlock.js

```diff
@@ -0,0 +1,27 @@
+import isEqual from 'lodash/isEqual'
+import { wrapActionsWithResolvedNode } from './utils'
+
+export const state = () => ({
+  microBlock: {}
+})
+
+export const mutations = {
+  setMicroBlock (state, microBlock) {
+    state.microBlock = microBlock
+  }
+}
+
+export const actions = wrapActionsWithResolvedNode({
+  async getMicroBlockFromHash ({ state, rootGetters: { node }, commit }, hash) {
+    const block = await node.api.getMicroBlockHeaderByHash(hash)
+    block.transactions = (await node.api.getMicroBlockTransactionsByHash(hash)).transactions
+
+    if (isEqual(state.block, block)) {
```

Comment:
why is this check required?

Reply:
It just prevents an extra commit to state. Now I think it's better if its not there.